src/Symfony/Component/Messenger/Tests/Transport/Sender/SendersLocatorTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Tests\Transport\Sender;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Messenger\Exception\RuntimeException;
+use Symfony\Component\Messenger\Tests\Fixtures\DummyMessage;
+use Symfony\Component\Messenger\Tests\Fixtures\SecondMessage;
+use Symfony\Component\Messenger\Transport\Sender\SendersLocator;
+use Symfony\Component\Messenger\Transport\Sender\SenderInterface;
+
+class SendersLocatorTest extends TestCase
+{
+    public function testItReturnsTheSenderBasedOnTheMessageClass()
+    {
+        $sender = $this->getMockBuilder(SenderInterface::class)->getMock();
+        $locator = new SendersLocator(array(
+            DummyMessage::class => array($sender),
+        ));
+
+        $this->assertSame(array($sender), iterator_to_array($locator->getSenders(DummyMessage::class)));
+        $this->assertSame(array(), iterator_to_array($locator->getSenders(SecondMessage::class)));
```

Comment:
it's technically not safe to call `iterator_to_array` on the result as it's return type is `iterable` which can be an array. I usually have a helper function for this `iterableToArray`

Reply:
that's a test case: we *do* expect an iterator. Even if it's implicit, at least it protects against changing the return value to an array.

Reply:
Then that should be a new assertion. A static code analyzer would complain about this code.